routing/blindedpath/blinded_path.go

```diff
@@ -435,11 +437,27 @@ func collectRelayInfo(cfg *BuildBlindedPathCfg, path *candidatePath) (
 			}
 		}
 
-		policy, err = cfg.AddPolicyBuffer(policy)
+		if policy.MinHTLCMsat > cfg.ValueMsat {
+			return nil, 0, 0, fmt.Errorf("minHTLC of hop policy "+
+				"larger than payment amt: sentAmt(%v), "+
+				"minHTLC(%v)", cfg.ValueMsat,
+				policy.MinHTLCMsat)
+		}
+
+		bufferPolicy, err := cfg.AddPolicyBuffer(policy)
 		if err != nil {
 			return nil, 0, 0, err
 		}
 
+		// We only use the new buffered policy if the new minHTLC value
+		// does not violate the sender amount.
+		//
+		// NOTE: We don't check this for maxHTLC, because the payment
+		// amount can always be splitted using MPP.
+		if bufferPolicy.MinHTLCMsat <= cfg.ValueMsat {
+			policy = bufferPolicy
```

Comment:
IIUC we should not choose this hop if the buffered policy cannot meet our requirement? I think we can do a single check on this `if bufferPolicy.MinHTLCMsat > cfg.ValueMsat {` and return a wrapped `errInvalidBlindedPath` here if `bufferPolicy.MinHTLCMsat` is at least `policy.MinHTLCMsat`

Reply:
yeah no strong opinion here (since i don't know it well😂), more like a question so feel free to ignore it.